skimage/transform/radon_transform.py

```diff
@@ -62,6 +67,8 @@ def radon(image, theta=None, circle=True):
     if theta is None:
         theta = np.arange(180)
 
+    image = convert_to_float(image, preserve_range)
```

Comment:
I think I would feel better with a deprecation cycle. This kind of unit change is REALLY annoying pin down, and a warning can go a long way.

Reply:
This deprecation warning should only be raised when a non float array is provided, right? But this is strange since the doc says that only float array is supported.

Reply:
The challenges encountered here are why this deserves its own PR. I would identify the cases where the behavior is changed, and warn only on those.

Finally, update the documentation specifying the support for uint8 if you want. If you don't, keep it as is.

Reply:
Yes, I agree, the addition of a new parameter can't be harmless. I will write a test to assert the multiple data types support.
The use of `convert_to_float` allows to use any data type, so I think that it is enough to simply remove the restriction to float in the doc.